### PR TITLE
Release/4.0

### PR DESCRIPTION
**Pull request by ingenthr**
Please review.  Note that the last one here has about an hour or two of work to go, so I'll update that Monday.  There are many, many examples.

Please don't merge the last one yet.

### CODE REVIEW

File: modules/howtos/examples/transactions-example.ts
Comment (osfameron) [inline modules/howtos/examples/transactions-example.ts]: this reference to *ottoman* doesn't look like it's used?

(and should be avoided until 4.1. in any case, as I understood it)

File: modules/howtos/examples/transactions-example.ts
Comment (osfameron) [inline modules/howtos/examples/transactions-example.ts:696]: this .catch/.then boilerplate is unneeded in later versions of Node.js, which handle these cases implicitly.

Are we best off keeping this in all examples until we have deprecated support for older versions?

File: modules/howtos/examples/transactions-example.ts
Comment (osfameron) [inline modules/howtos/examples/transactions-example.ts]: phew!

**Review by osfameron — COMMENTED** [review]
Looks good so far!
I'll have a go now running the examples and adding them to our doc-example tests.

File: modules/howtos/examples/transactions-example.ts
Comment (ingenthr) [reply modules/howtos/examples/transactions-example.ts:696]: I'd defer to you folks.  I think if Node 16 does it correctly, we can update the docs and let users "backport" if needed.  I just copied this over from the typescript example in the repo to be consistent.

**Review by RichardSmedley — APPROVED** [review]
LGTM
We can tackle the Java-centric common content later this week...

Reviewing as approved to be ready for tomorrow, but please feel free to keep tweaking where necessary.  :+1:

**Comment by ingenthr** [comment]
Please give this another look @osfameron and @RichardSmedley.  I'll admit that there are quite a few TODOs in there that will need to get turned into specific issues.  That said, I think it's complete enough for this dot-oh release.  It's quite complete.

File: modules/howtos/examples/transactions-example.ts
Comment (osfameron) [inline modules/howtos/examples/transactions-example.ts:22]: This reminds me needs to be added to `test-howtos.bats`
Doesn't need to block merge, but I'll try to do that today.

File: modules/howtos/examples/transactions-example.ts
Comment (osfameron) [inline modules/howtos/examples/transactions-example.ts:299]: what's the purpose of this commented code?

File: modules/howtos/examples/transactions-example.ts
Comment (osfameron) [inline modules/howtos/examples/transactions-example.ts:366]: another example of commented out code - I think this needs to either be removed, or marked with a TODO if still relevant?

**Review by osfameron — COMMENTED** [review]
Looks good!

There is quite a lot of commented out code though, some of which *looks like* it's been fully incorporated and should therefore be deleted?

(I'm happy with commented code next to a `TODO` comment of course!)

I'd prefer those to be addressed, mostly as if Matt moves on to something else after this release, the code will never get cleaned up... but not sure this requires a full "Request changes" so, I guess 👍

File: modules/howtos/examples/transactions-example.ts
Comment (ingenthr) [reply modules/howtos/examples/transactions-example.ts:366]: indeed.  I forgot that before I was going to push to you, I was going to revisit all of the tagged code samples and remove any extraneous commented code.

You might ask why I had it there at all?  I'd like to get @brett19's review and it's substantially more convenient to see the Java code there side by side rather than needing to dig it up.

I'll push a change for this in a bit.

**Pull request merged**
